encoder/vaapiencoder_base.cpp

```diff
@@ -574,10 +582,26 @@ bool VaapiEncoderBase::initVA()
     }
 
     if (RATE_CONTROL_NONE != m_videoParamCommon.rcMode) {
-        attrib.type = VAConfigAttribRateControl;
-        attrib.value = m_videoParamCommon.rcMode;
-        pAttrib = &attrib;
+        attrib[0].type = VAConfigAttribRateControl;
+        attrib[0].value = m_videoParamCommon.rcMode;
+        pAttrib = attrib;
         attribCount = 1;
+#ifdef ENABLE_HEVC_ENC_ON_STUDIO_VA
+        /*
+        It's necessary to pass two attribute:
+        VAConfigAttribRTFormat and VAConfigAttribRateControl to libva;
+        The value of VAConfigAttribRateControl should be 0x82 not RATE_CONTROL_CBR;
+        Or else, HEVC encoding will end up with an error: attribute not supported.
+        */
+        if (RATE_CONTROL_CBR == m_videoParamCommon.rcMode) {
+            attrib[0].type = VAConfigAttribRTFormat;
+            attrib[0].value = 0;
+            attrib[1].type = VAConfigAttribRateControl;
+            attrib[1].value = 0x82; //RATE_CONTROL_CBR
```

Comment:
could you find where we need this magic number, is it another RATE_CONTROL_XXX?

Reply:
I can't find out any MACRO for it.

Reply:
is it a VA_RC_MB|VA_RC_CBR ?

Reply:
I get your mean, thanks for your help.